ArduPlane/mode_LoiterAltQLand.cpp

```diff
@@ -10,17 +10,17 @@ bool ModeLoiterAltQLand::_enter()
         return true;
     }
 
+    const Location previous_wp = plane.next_WP_loc;
+    const bool already_in_a_loiter = plane.reached_loiter_target();
```

Comment:
this could be true in modes like FBWA, cruise etc, would go to the wrong position

Reply:
@tridge  changed to
`const bool already_in_a_loiter = plane.previous_mode->does_auto_navigation() && plane.reached_loiter_target();`

Which I think should handle all the cases.
Note, I CAN NOT use the most accurate and direct check `plane.loiter.start_time_ms > 0` because it gets set to 0 in set_mode() before _enter() is called. I looked through the code and it looks like moving that set to 0 is probably OK to move to after the _enter() is successful but that has the potential to create some gremlins that aren't worth it. At this point I feel like reached_loiter_target() not being set to zero in the same place is a bug fixing that by setting to zero would break this PR. To globally fix that we would change reached_loiter_target() to `return control_mode->does_auto_navigation() && nav_controller->reached_loiter_target();` but that would need some additional testing.